lib/portage/dbapi/vartree.py

```diff
@@ -5800,28 +5802,33 @@ def mergeme(
                 # whether config protection or not, we merge the new file the
                 # same way.  Unless moveme=0 (blocking directory)
                 if moveme:
-                    # Create hardlinks only for source files that already exist
-                    # as hardlinks (having identical st_dev and st_ino).
-                    hardlink_key = (mystat.st_dev, mystat.st_ino)
+                    # only replace the existing file if it differs, see #722270
+                    already_merged = os.path.exists(mydest)
+                    if already_merged and filecmp.cmp(mysrc, mydest, shallow=False):
```

Comment:
I see that filecmp.cmp does not compare xattrs, which could be an issue given that movefile copies xattrs.

Reply:
I tackled the xattrs handling and file mode comparison in 3ff0d5a. The xattr comparison has been done in a similar way as `_copyxattr(...)` to respect `PORTAGE_XATTR_EXCLUDE`.